pig-tests/test/org/partiql/pig/tests/PermuteTransformTests.kt

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package org.partiql.pig.tests
+
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.fail
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.ArgumentsSource
+import org.partiql.pig.tests.generated.TestPermuteDomainA
+import org.partiql.pig.tests.generated.TestPermuteDomainAToTestPermuteDomainBVisitorTransform
+import org.partiql.pig.tests.generated.TestPermuteDomainB
+
+/**
+ * Test cases for the 5 basic scenarios for domain permutation.
+ *
+ * test_permute_domain_b is a permutation of test_permute_domain_a.  See test_permute_domain_a and
+ * test_permute_domain_b in sample-universe.ion for more details
```

Comment:
nit: for readability, could these be shortened? The cross-domain transform `TestPermuteDomainAToTestPermuteDomainBVisitorTransform` is pretty lengthy.

Perhaps?: 
- `test_permute_domain_a` -> `domain_a`
- `test_permute_domain_b` -> `domain_b`

Reply:
that is a mouthful--next push